test/e2e/install_test.go

```diff
@@ -0,0 +1,210 @@
+	scriptURL string // To test a non-published script
+	noFlush   bool   // To prevent eventual cleanup, to test install_script won't override existing configuration
+)
+
+// note: no need to call flag.Parse() on test code, go test does it
+func init() {
+	flag.StringVar(&flavor, "flavor", "datadog-agent", "defines agent install flavor")
+	flag.StringVar(&mode, "mode", "install", "test mode")
+	flag.BoolVar(&noFlush, "noFlush", false, "To prevent eventual cleanup, to test install_script won't override existing configuration")
+	flag.StringVar(&apiKey, "apiKey", os.Getenv("DD_API_KEY"), "Datadog API key")
+	flag.StringVar(&scriptURL, "scriptURL", defaultScriptURL, fmt.Sprintf("Defines the script URL, default %s", defaultScriptURL))
+}
+
+type linuxInstallerTestSuite struct {
+	e2e.Suite[e2e.VMEnv]
+}
+
+func TestLinuxInstallerSuite(t *testing.T) {
+	scriptType := "production"
+	if scriptURL != defaultScriptURL {
+		scriptType = "custom"
+	}
+	t.Run(fmt.Sprintf("We will %s %s with %s install_script on Ubuntu 22.04", mode, flavor, scriptType), func(t *testing.T) {
+		testSuite := &linuxInstallerTestSuite{}
+		e2e.Run(t,
+			testSuite,
+			e2e.EC2VMStackDef(ec2params.WithOS(ec2os.UbuntuOS)),
+			params.WithStackName(fmt.Sprintf("%s-%s-ubuntu22", mode, flavor)),
+		)
+	})
+}
+
+func (s *linuxInstallerTestSuite) TestInstallerScript() {
+	t := s.T()
+	vm := s.Env().VM
+	// Flavor selection
+	baseName := "datadog-agent"
+	configFile := "datadog.yaml"
+	if flavor == "datadog-dogstatsd" {
+		baseName = "datadog-dogstatsd"
+		configFile = "dogstatsd.yaml"
+	}
+
+	// Installation
+	if mode == "install" {
+		t.Log("Install latest Agent 7 RC")
+		cmd := fmt.Sprintf("DD_AGENT_FLAVOR=%s DD_AGENT_MAJOR_VERSION=7 DD_API_KEY=%s DD_SITE=\"datadoghq.com\" DD_REPO_URL=datad0g.com DD_AGENT_DIST_CHANNEL=beta bash -c \"$(curl -sL %s/install_script_agent7.sh)\"", flavor, apiKey, scriptURL)
+		vm.Execute(cmd)
+	} else if mode == "upgrade7" {
+		t.Log("Install latest Agent 7")
+		cmd := fmt.Sprintf("DD_AGENT_FLAVOR=%s DD_AGENT_MAJOR_VERSION=7 DD_API_KEY=%s bash -c \"$(curl -L %s/install_script_agent7.sh)\"", flavor, apiKey, scriptURL)
+		vm.Execute(cmd)
+		t.Log("Install latest Agent 7 RC")
+		cmd = fmt.Sprintf("DD_AGENT_FLAVOR=%s DD_AGENT_MAJOR_VERSION=7 DD_API_KEY=%s DD_REPO_URL=datad0g.com DD_AGENT_DIST_CHANNEL=beta bash -c \"$(curl -L %s/install_script_agent7.sh)\"", flavor, apiKey, scriptURL)
+		vm.Execute(cmd)
+	} else if mode == "upgrade6" {
+		t.Log("Install latest Agent 6")
+		cmd := fmt.Sprintf("DD_AGENT_FLAVOR=%s DD_AGENT_MAJOR_VERSION=6 DD_API_KEY=%s bash -c \"$(curl -L %s/install_script_agent6.sh)\"", flavor, apiKey, scriptURL)
+		vm.Execute(cmd)
+		t.Log("Install latest Agent 7 RC")
+		cmd = fmt.Sprintf("DD_AGENT_FLAVOR=%s DD_AGENT_MAJOR_VERSION=7 DD_API_KEY=%s DD_REPO_URL=datad0g.com DD_AGENT_DIST_CHANNEL=beta bash -c \"$(curl -L %s/install_script_agent7.sh)\"", flavor, apiKey, scriptURL)
```

Comment:
I feel like we have a lot of code that could be reused. That is shared between upgrade scenario. What do you think about creating a function to have something like
```go
if mode == "upgrade6" {
    upgradeToLatestRCFrom(6)
} else if mode == "upgrade7" {
    upgradeToLatestRCFrom(7)
}
```

Reply:
Oh this, there is a lot of refactoring incoming. I kept this PR to the smallest change set as possible, keeping code 1:1 with the original test script. Target is https://github.com/DataDog/agent-linux-install-script/pull/98